lib/tasks_private/instance_types.rake

```diff
@@ -1,215 +1,61 @@
-      :instance_store_size     => #{instance[:instance_store_size] > 0 ? (instance[:instance_store_size] / Numeric::GIGABYTE).to_s + '.gigabytes' : 0},
-      :instance_store_volumes  => #{instance[:instance_store_volumes]},#{instance[:instance_store_ssd] ? ' # SSD' : ''}
-      :architecture            => #{instance[:architecture]},
-      :virtualization_type     => #{instance[:virtualization_type]},
-      :network_performance     => :#{instance[:network_performance]},
-      :physical_processor      => "#{instance[:physical_processor]}",
-      :processor_clock_speed   => "#{instance[:processor_clock_speed]}",
-      :intel_aes_ni            => #{instance[:intel_aes_ni] ? 'true' : 'nil'},
-      :intel_avx               => #{instance[:intel_avx] ? 'true' : 'nil'},
-      :intel_avx2              => #{instance[:intel_avx2] ? 'true' : 'nil'},
-      :intel_turbo             => #{instance[:intel_turbo] ? 'true' : 'nil'},
-      :ebs_optimized_available => #{instance[:ebs_optimized_available] ? 'true' : 'nil'},
-      :enhanced_networking     => #{instance[:enhanced_networking] ? 'true' : 'nil'},
-      :cluster_networking      => #{instance[:cluster_networking]  ? 'true' : 'nil'},
-      :vpc_only                => #{instance[:vpc_only]},
-    },
-    EOT
-  end
-
-  def cluster_networking?(type)
-    # https://aws.amazon.com/ec2/instance-types/#Cluster_Networking
-    type.split('.').first.in?(%w(r4 x1 m4 c4 c3 i2 cr1 g2 hs1 p2 d2))
-  end
-
-  def format_instance_miq(i)
-    {}.tap do |o|
-      # x1.16large manually fixed to x1.16xlarge
-      # i3.16large manually fixed to i3.16xlarge
-      %i(family instance_store_size instance_store_volumes vpc_only enhanced_networking).each do |key|
-        o[key] = i[key]
+# frozen_string_literal: true
+
+# https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/using-ppslong.html#download-the-offer-index
+
+require 'pry-byebug'
+require 'awesome_print'
+
+namespace 'aws:extract' do
+  desc 'Get / renew instance types and details list from AWS Price List Bulk API'
+  task :instance_types do
+    require 'uri'
+    require 'set'
+    require 'json'
+    require 'yaml'
+    require 'net/http'
+
+    require_relative 'lib/aws_products_data_collector'
+    require_relative 'lib/aws_instance_types_parser'
+    require_relative 'lib/simple_logging'
+
+    data_dir = ManageIQ::Providers::Amazon::Engine.root.join('db/fixtures')
+    data_dir.mkpath
+
+    out_file = data_dir.join('aws_instance_types.yml')
+    out_file_old = data_dir.join('aws_instance_types_old.yml')
+
+    # 'Dedicated Host', # TODO: do we need bare metal types here? e.g. "m5", "p3", etc.
+    products = AwsProductsDataCollector.new(
+      'AmazonEC2', 'Compute Instance', AwsInstanceTypesParser::REQUIRED_ATTRIBUTES,
+      'instanceType', 'currentGeneration').products_list
+
+    instances = AwsInstanceTypesParser.new(products).instances_list
+
+    # prevent yaml anchors/aliases
```

Comment:
This should not be necessary. To avoid aliases, there are multiple approaches.  If you are creating the data structures in question, then you can just avoid putting references into other container instances (e.g. don't put the same array into multiple hashes).  Instead, use `.dup` or `.clone`.  If you aren't creating the hashes, and instead they come from Amazon, you can use `.deep_clone` or `Marshal.load(Marshal.dump(obj)` which will ultimately serialize and then deserialize, which should give you what you need